include/zenoh/api/reply.hxx

```diff
@@ -33,6 +33,13 @@ class ReplyError : public Owned<::z_owned_reply_err_t> {
     const Bytes& get_payload() const {
         return interop::as_owned_cpp_ref<Bytes>(::z_reply_err_payload(interop::as_loaned_c_ptr(*this)));
     }
+#if defined(ZENOHCXX_ZENOHC)
+    /// @brief The payload of this error.
+    /// @return error payload.
+    Bytes& get_payload() {
```

Comment:
today there is no way to get mutable ReplyErr or Sample from reply, so this method is useless as it is.

Reply:
Then we should add this. SHM buffer may reside literally everywhere where ZBytes is

Reply:
That is the problem. We can not add these methods, because this would allow user to apply std::move to mutable Sample or ReplyError which would lead to their destruction. Since on zenoh-c level calling destructors is only possible on owned objects and sample/error are stored as loaned inside reply this would lead to UB. 

So it is ok to merge this pr as is, but adding mutable access to sample/reply_err from Reply would require first resolving https://github.com/eclipse-zenoh/zenoh-c/pull/718.

Reply:
Made mutable access to sample/reply_err from Reply private for a while.